go/api/base_client.go

```diff
@@ -69,14 +79,189 @@ func failureCallback(channelPtr unsafe.Pointer, cErrorMessage *C.char, cErrorTyp
+
+	case Subscribe, PSubscribe, SSubscribe, Unsubscribe, PUnsubscribe, SUnsubscribe:
+		if len(arrayValues) < 2 {
+			return
+		}
+		channel, ok := utils.ToString(arrayValues[0])
+		if !ok {
+			return
+		}
+		msgContent, ok := utils.ToString(arrayValues[1])
+		if !ok {
+			return
+		}
+		message = NewPubSubMessage(msgContent, channel)
+
+	default:
+		// log unsupported push kind
+
+	}
+
+	if clientPtr != nil {
+		// Look up the client in our registry using the pointer address
+		ptrValue := uintptr(clientPtr)
+		client := GetClientByPtr(ptrValue)
+
+		if client != nil {
+			// If the client has a message handler, use it
+			if handler := client.GetMessageHandler(); handler != nil {
+				// Create PushInfo and pass it to the handler
+				pushMsg := PushInfo{
+					Kind:    pushKind,
+					Message: message,
+				}
+				handler.Handle(pushMsg)
+			}
+		} else {
+			fmt.Printf("Client not found for pointer: %v\n", ptrValue)
+		}
+	}
+}
+
+// Helper function to process a CommandResponse array into a Go slice
+func processCommandResponseArray(cResponse *C.struct_CommandResponse) ([]any, error) {
+	if typeErr := checkResponseType(cResponse, C.Array, false); typeErr != nil {
+		return nil, typeErr
+	}
+
+	arrayLen := int(cResponse.array_value_len)
+	arrayPtr := cResponse.array_value
+	result := make([]any, arrayLen)
+
+	// Iterate through the array elements
+	for i := 0; i < arrayLen; i++ {
+		element := unsafe.Pointer(uintptr(unsafe.Pointer(arrayPtr)) + uintptr(i)*unsafe.Sizeof(*arrayPtr))
+		elemPtr := (*C.struct_CommandResponse)(element)
+
+		// Convert element based on type
+		switch elemPtr.response_type {
+		case C.String:
+			strLen := int(elemPtr.string_value_len)
+			if strLen > 0 && elemPtr.string_value != nil {
+				bytes := C.GoBytes(unsafe.Pointer(elemPtr.string_value), C.int(strLen))
+				result[i] = bytes
+			} else {
+				result[i] = []byte{}
+			}
+		case C.Int:
+			result[i] = int64(elemPtr.int_value)
+		case C.Float:
+			result[i] = float64(elemPtr.float_value)
+		case C.Bool:
+			result[i] = bool(elemPtr.bool_value)
+		case C.Null:
+			result[i] = nil
+		default:
+			// For other types, we'd need more complex handling
+			// For simplicity, convert to string representation
+			result[i] = fmt.Sprintf("Unsupported type: %d", elemPtr.response_type)
+		}
+	}
+	return result, nil
+}
+
 type clientConfiguration interface {
 	toProtobuf() (*protobuf.ConnectionRequest, error)
 }
 
 type baseClient struct {
-	pending    map[unsafe.Pointer]struct{}
-	coreClient unsafe.Pointer
-	mu         sync.Mutex
+	pending        map[unsafe.Pointer]struct{}
+	coreClient     unsafe.Pointer
+	mu             sync.Mutex
+	messageHandler *MessageHandler
+}
+
+func (client *baseClient) GetClientID() string {
+	return fmt.Sprintf("%p", client)
+}
+
+// SetMessageHandler assigns a message handler to the client for processing pub/sub messages
+func (client *baseClient) SetMessageHandler(handler *MessageHandler) {
+	client.messageHandler = handler
+}
```

Comment:
Not needed anymore as I understand

Reply:
GetClientID is removed, however the SetMessageHandler is still used.